data/src/main/java/org/apache/iceberg/data/InternalRecordWrapper.java

```diff
@@ -80,10 +80,15 @@ public int size() {
 
   @Override
   public <T> T get(int pos, Class<T> javaClass) {
-    if (transforms[pos] != null) {
-      return javaClass.cast(transforms[pos].apply(wrapped.get(pos, Object.class)));
+    Object value = wrapped.get(pos, Object.class);
```

Comment:
I don't think this should change the behavior of the else case where the java class is passed to `wrapped.get`. For that case, the right thing to do is to delegate the operation to `wrapped`. The reason to use `Object.class` is when we have a transform function and know that the java class passed in won't work.